### DISCUSSION TITLE
Problem with typehinting methods of interface

### DISCUSSION THREAD

**Discussion by stalkerxxl** [symfony/symfony#47121] [Q&A] [answered]
Hi all!
I recently found a problem: typehinting isn't working at class's methods, which implement interface. I tested it on a clean Symfony and PHPStorm install.

```php
<?php

namespace App\Controller;

use Psr\Log\LoggerInterface;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;

class HomeController extends AbstractController
{
    #[Route('/home', name: 'app_home')]
    public function index(LoggerInterface $phpLogger): Response
    {
	$phpLogger->error('test error');

        return $this->render('home/index.html.twig', [
            'controller_name' => 'HomeController',
        ]);
    }
}
```

When i'm clicked **error()** method - PHPStorm can't detect interface's implementation. [Weird, but even in the documentaion is a similar example - and everything in there is fine](https://symfony.com/doc/current/logging.html#logging-a-message)
![phpstorm64_HjjFUkbpOB](https://user-images.githubusercontent.com/19319937/181781945-8d80c9da-9ae8-49af-b070-9220a4d3bb83.png)
![hCEXRoEkXg](https://user-images.githubusercontent.com/19319937/181781951-877c65e7-7e5f-4b26-9124-135d4d00bc98.png)
![phpstorm64_EsIsshdLZF](https://user-images.githubusercontent.com/19319937/181782034-e896efff-b99a-4e46-9801-c40722571a63.png)
Same problem with Http client (when trying to inject HttpClientInterface)

**I use:**
Symfony 6.1.3 (default config)
PHPStorm 2022.1.4
PHP 8.1.7

_Sorry for my english (my native - ukranian)_

**Comment by lyrixx** [comment, marked as answer]
This is an issue with PHPStorm here. Not Symfony :)

**Reply by stalkerxxl** [reply]
Yes, you right..
https://youtrack.jetbrains.com/issue/WI-65838/Prefer-stubs-over-definitions-from-vendor-from-the-entire-project-too
